### ISSUE TITLE
Update symbolic models to cover early auth tag in RespHello

### ISSUE THREAD

**Issue by koraa** [rosenpass/rosenpass#169]
See #168 

/bounty 100

**Comment by Min2who** [comment]
@koraa can you give more context about the issue

**Comment by lafamumin** [comment]
@koraa Can you please elaborate more ?

**Comment by AliceOrunitia** [comment]
@pqcfox Hi! Just wanted to check in and ask how you are getting on with this and if you have any questions that have come up or if there's any help we can provide?

**Comment by digital-phoenix** [comment]
@pqcfox would you like any help with this issue?

**Comment by pqcfox** [comment]
@AliceOrunitia @digital-phoenix my dearest apologies! I got started but had to halt just because of a lack of time--I'm picking it back up today, will share any questions I run into. 

Thanks for checking in!!

**Comment by pqcfox** [comment]
Awesome--I think I'm all caught up on the background stuff!

Just to make sure I'm on the right track, for the auth tag, will it work (on the analysis side) just to add a copy of RHR7 to the top of RespHello, i.e. just by copying the corresponding RHR7 lines in `protocol.mpv` to the top of `RESPHELLO_PRODUCE()` and `RESPHELLO_CONSUME()`? The only mentions of auth tags I could find in the whitepaper were just these `AEAD::enc(empty())` responses, so I assumed these were what @koraa was intending in #168.

If so, should I renumber all of the RHR steps in the comments, assuming the change to #168 will do the same? (I'm happy to help out with that one after this as well if there's room to.)

(Lastly, @digital-phoenix should I assume that analysis of the DOS exhaustion mentioned in #168 will be handled by #167? Thanks!)

**Comment by digital-phoenix** [comment]
@pqcfox the current auth variable is created from the chaining key and the chaining key is built from each preceding operation in the process. We don't want to move RHR7 as it would change the data that the auth variable contains. @koraa would need to let us know the exact details of the early auth tag as it is a new piece of data.

#168 will handle all of the DOS exhaustion properties from the cookie mechanism but it sounds like the early auth tag will provide additional protection from Dos exhaustion. Which will require new analysis code to show that extra layer of protection.

**Comment by pqcfox** [comment]
Makes sense on both fronts--thanks so much! In that case, I'll wait for the update on that front and pick up once the auth tag contents are determined.

**Comment by pqcfox** [comment]
In the meantime, any thoughts of what would be reasonable to include, just so I can try implementing something for now?

**Comment by digital-phoenix** [comment]
I don't have any suggestions for other practical inclusions. But to get used to the code I'd try taking a look at the queries and lemmas in 01_secrecy.entry.mpv and experiment with adding new ones.

**Comment by koraa** [comment]
@pqcfox @digital-phoenix @AliceOrunitia @prabhpreet

This issue is a bit too complex for me to specify in a top down manner; we have a couple people interested in symbolic analysis on rosenpass and the work surrounding that.

There are papers planned, grant proposals, more modeling tasks, and a some initiatives to make modeling more ergonomic.

Then we can take the notes and write them down here.

How about we just set up a meeting to talk about coordinating this effort? I am interested to learn what your motivations are to work on this and how to cater to this; how you would like to be involved. My next two weeks are blocked, but there are some slots on Friday, Feb. 2. 14:10 CET (13:10 UTC) would suit me very well.

I don't think funding will be a major issue; there is enough money to pay for peoples involvement.

**Comment by pqcfox** [comment]
Sounds excellent to me! I'm on UTC-8, so if there's anything later in the day that'd work even better, but I'm more than happy to be there at 14:10 CET.

**Comment by digital-phoenix** [comment]
I'd also prefer a slightly later time if possible

**Comment by koraa** [comment]
@pqcfox @digital-phoenix We could do 16:10 or 17:10 CET?

**Comment by pqcfox** [comment]
17:10 CET is perfect for me!

**Comment by digital-phoenix** [comment]
16:10 and 17:10 both work for me

**Comment by pqcfox** [comment]
@AliceOrunitia that's perfect, thanks so much!

**Comment by ilka-schulz** [comment]
We are closing this issue for now and might attempt it later. We received several low-quality LLM PRs for our algora-bounty issues and need to work through a huge amount of slop first.
